src/sig.c

```diff
@@ -93,4 +94,26 @@ void install_rotate_signal_handlers()
 
   /* This one triggers a log rotate. */
   install(SIGUSR1, rotate_log_signal_handler);
+
+  if (arg_journal_rotate_interval) {
```

Comment:
I was curious as to what happens if you specify a negative rotation interval, but it looks like some platforms will return EINVAL here and others will happily do something.  There's definitely a "doctor, doctor, it hurts when I do this" situation about that case, but you could always require a nonnegative value.